### DISCUSSION TITLE
Thoughts on Modiftying gpio-line-names for gpiod? ( / where can I find the .dts files?)

### DISCUSSION THREAD

**Discussion by conradmenchine** [OE4T/meta-tegra#1691] [Q&A] [answered]
Hi All,
I'm looking to add / modify some of the gpio-line-names properties in the device tree for use with gpiod. Unpacking the built `tegra234-p3701-0000-p3737-0000-<MACHINE>.dtb` file found deep inside the work dir, I was able to find that the GPIO already has some things assigned to gpio-line-names:
```
gpio@2200000 {
		compatible = "nvidia,tegra234-gpio";
		reg-names = "security\0gpio";
		reg = <0x00 0x2200000 0x00 0x10000 0x00 0x2210000 0x00 0x10000>;
		interrupts = <0x00 0x120 0x04 0x00 0x121 0x04 0x00 0x122 0x04 0x00 0x123 0x04 0x00 0x124 0x04 0x00 0x125 0x04 0x00 0x126 0x04 0x00 0x127 0x04 0x00 0x128 0x04 0x00 0x129 0x04 0x00 0x12a 0x04 0x00 0x12b 0x04 0x00 0x12c 0x04 0x00 0x12d 0x04 0x00 0x12e 0x04 0x00 0x12f 0x04 0x00 0x130 0x04 0x00 0x131 0x04 0x00 0x132 0x04 0x00 0x133 0x04 0x00 0x134 0x04 0x00 0x135 0x04 0x00 0x136 0x04 0x00 0x137 0x04 0x00 0x138 0x04 0x00 0x139 0x04 0x00 0x13a 0x04 0x00 0x13b 0x04 0x00 0x13c 0x04 0x00 0x13d 0x04 0x00 0x13e 0x04 0x00 0x13f 0x04 0x00 0x140 0x04 0x00 0x141 0x04 0x00 0x142 0x04 0x00 0x143 0x04 0x00 0x144 0x04 0x00 0x145 0x04 0x00 0x146 0x04 0x00 0x147 0x04 0x00 0x148 0x04 0x00 0x149 0x04 0x00 0x14a 0x04 0x00 0x14b 0x04 0x00 0x14c 0x04 0x00 0x14d 0x04 0x00 0x14e 0x04 0x00 0x14f 0x04>;
		gpio-controller;
		#gpio-cells = <0x02>;
		interrupt-controller;
		#interrupt-cells = <0x02>;
		gpio-ranges = <0x2f9 0x00 0x00 0x8a 0x2f9 0x8a 0x92 0x1a>;
		status = "okay";
		gpio-line-names = <0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x4750494f 0x33350000 0x00 0x493253 0x325f5343 0x4c4b0049 0x3253325f 0x444f5554 0x493253 0x325f4449 0x4e004932 0x53325f46 0x53000000 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x4750494f 0x32370000 0x00 0x00 0x475049 0x4f313700 0x00 0x00 0x4d434c 0x4b303500 0x50574d 0x30310000 0x00 0x00 0x00 0x00 0x00 0x00 0x535049 0x315f434c 0x4b005350 0x49315f4d 0x49534f00 0x53504931 0x5f4d4f53 0x49005350 0x49315f43 0x53305f4e 0x535049 0x315f4353 0x315f4e00 0x00>;
		phandle = <0x04>;

		camera-control-output-low {
			gpio-hog;
			output-low;
			gpios = <0x3b 0x00 0x3e 0x00 0xa1 0x00 0xa0 0x00>;
			label = "cam0-rst\0cam0-pwdn\0cam1-rst\0cam1-pwdn";
		};
	};
```

To save you the suspense, those unpack to:
```














GPIO
35

I2S
2_SC
LKI
2S2_
DOUT
I2S
2_DI
NI2
S2_F
S









GPIO
27


GPI
O17


MCL
K05
PWM
01






SPI
1_CL
KSP
I1_M
ISO
SPI1
_MOS
ISP
I1_C
S0_N
SPI
1_CS
1_N
```
So, it looks like the SPI and I2S pins are named, but everything else is not. Strangely, I haven't been able to find the source files that assign these names in order to modify/extend them. Has anyone here gone through this process before?

Thanks!

**Comment by kekiefer** [comment]
What BSP version are you using? For R36, the stock devicetrees reside in the `nvidia-kernel-oot` package, and the original gpio node is at `/bus@0/gpio@2200000` in `hardware/nvidia/t23x/nv-public/tegra234.dtsi` -- where there is no `gpio-line-names` property set. For R35 and before, the stock devicetrees reside in the kernel sources.

There's nothing stopping you from adding them. You can do so in any included dts file by adding a node with that same path, or by referencing the `&gpio { ...` label, and the devicetree compiler will concatenate any additional node properties as it sees them. You would accomplish this by either making a [custom devicetree package](https://github.com/OE4T/tegra-demo-distro/blob/master/layers/meta-tegrademo/recipes-bsp/tegrademo-devicetree/tegrademo-devicetree_1.0.bb) or by patching the original sources (the `devtool modify ...` workflow makes this pretty easy).

**Reply by conradmenchine** [reply]
Thank you for your response! I'm building with Kirkstone, and I found the tegra234-soc-base.dtsi file, which seems to define some of the gpio stuff. My main confusion is that I can't find any .dtsi files that have the `gpio-line-names` property populated, so it's a bit unclear how it's getting into the final dtb. My understanding is that I'm going to need to replace the whole  `gpio-line-names` property, so finding one to modify rather than having to reverse-engineer the thing back into strings would be ideal

**Reply by madisongh** [reply, marked as answer]
A quick grep turned up [this](https://github.com/OE4T/linux-tegra-5.10/blob/db857ecdbe10de229a3cf125831afd6226417eac/nvidia/platform/t23x/concord/kernel-dts/cvb/tegra234-p3737-0000-a04.dtsi#L347-L374), is that what you're looking for?

**Reply by conradmenchine** [reply]
Thank you Matt! You've saved me once again, That's exactly what I was looking for!
